src/test/java/io/odh/test/e2e/standard/PipelineV2ServerST.java

```diff
@@ -0,0 +1,401 @@
+                        .endPersistenceAgent()
+                        .withNewScheduledWorkflow()
+                            .withCronScheduleTimezone("UTC")
+                            .withDeploy(true)
+                        .endScheduledWorkflow()
+                    .endSpec()
+                    .build();
+            ResourceManager.getInstance().createResourceWithWait(dspa);
+        });
+
+        Allure.step("Wait for Pipeline API server to come up");
+        Resource<Endpoints> endpoints = client.endpoints().inNamespace(prjTitle).withName("ds-pipeline-pipelines-definition");
+        waitForEndpoints(endpoints);
+
+        Allure.step("Connect to the API server");
+        Resource<Route> route = ocClient.routes()
+                .inNamespace(prjTitle).withName("ds-pipeline-pipelines-definition");
+
+        // TODO(jdanek) I still don't know how to do oauth, so let's forward a port
+        ServiceResource<Service> svc = client.services().inNamespace(prjTitle).withName("ds-pipeline-pipelines-definition");
+        try (LocalPortForward portForward = svc.portForward(8888, 0)) {
+            KFPv2Client kfpClient = new KFPv2Client("http://localhost:%d".formatted(portForward.getLocalPort()));
+
+            // WORKAROUND(RHOAIENG-3250): delete sample pipeline present on ODH
+            deletePreexistingPipelinesAndVersions(kfpClient);
+
+            KFPv2Client.Pipeline importedPipeline = kfpClient.importPipeline(pipelineTestName, pipelineTestDesc, pipelineTestFilepath);
+
+            List<KFPv2Client.Pipeline> pipelines = kfpClient.listPipelines();
+            assertThat(pipelines.stream().map(p -> p.displayName).collect(Collectors.toList()), Matchers.contains(pipelineTestName));
+
+            Map<String, Object> parameters = Map.of(
+                    "min_max_scaler", false,
+                    "neighbors", 1,
+                    "standard_scaler", true
+            );
+            KFPv2Client.PipelineRun pipelineRun = kfpClient.runPipeline(pipelineTestRunBasename, importedPipeline.pipelineId, parameters, "Immediate");
+
+            kfpClient.waitForPipelineRun(pipelineRun.runId);
+
+            List<KFPv2Client.PipelineRun> statuses = kfpClient.getPipelineRunStatus();
+            assertThat(statuses.stream()
+                    .filter(run -> run.runId.equals(pipelineRun.runId))
+                    .map(run -> run.state)
+                    .findFirst().orElseThrow(), Matchers.is("SUCCEEDED"));
+
+            checkPipelineRunK8sDeployments(prjTitle, pipelineRun.runId);
+
+            kfpClient.deletePipelineRun(pipelineRun.runId);
+            for (KFPv2Client.PipelineVersion pipelineVersion : kfpClient.listPipelineVersions(importedPipeline.pipelineId)) {
+                kfpClient.deletePipelineVersion(importedPipeline.pipelineId, pipelineVersion.pipelineVersionId);
+            }
+            kfpClient.deletePipeline(importedPipeline.pipelineId);
+        }
+    }
+
+    @io.qameta.allure.Step
```

Comment:
Shouldn't we rename `Step` annotation for test metadata to something like `TestStep` to avoid these lines?

Reply:
I don't particularly mind writing this. On the other hand, it's good to have the skodjob Step annotation have short name, because the lines with it tend to get long.

Reply:
skodjob Step?  you mean test-metadata-generator ?

Reply:
Guess we can discuss it offline